src/Entity/EntityFactory.php

```diff
@@ -133,14 +142,42 @@ public static function createPackage(array $data = []): Package
 
     /**
      * @param array $data
-     * @return Transaction
+     *
+     * @return mixed
+     *
+     * @throws AdsException
      */
-    public static function createTransaction(array $data = []): Transaction
+    public static function createTransaction(array $data = [])
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 9, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1142)